prov/psm2/src/psmx2.h

```diff
@@ -651,8 +651,8 @@ struct psmx2_fid_cntr {
 	struct psmx2_fid_domain	*domain;
 	int			events;
 	uint64_t		flags;
-	atomic_t		counter;
-	atomic_t		error_counter;
+	atomic32_t		counter;
+	atomic32_t		error_counter;
```

Comment:
these counters are followed the previous implementation and are stayed signed values (int32_t). Did you mean that is it will be good if we add new atomic unsigned datatype?